### ISSUE TITLE
INVISIBLE_RECAPTCHA_DEBUG=true doesn't seem to work

### ISSUE THREAD

**Issue by david-j-davis** [albertcht/invisible-recaptcha#76]
Currently on my site the invisible recaptcha is not working on my registration page, and upon submit, the recaptca is failing and I have INVISIBLE_RECAPTCHA_DEBUG=true but don't see any logs inside of laravel.log or console errors. Where does this debug show up?

**Comment by johnnymast** [comment]
sir i have gave a big time issue report it wasn't taken serious so i think we are wasting our time. My long report is here (about a slightly diffrent issue) https://github.com/albertcht/invisible-recaptcha/issues/70

**Comment by albertcht** [comment]
Hi @david-j-davis ,

If you set `INVISIBLE_RECAPTCHA_DEBUG` as true, you should be able to see the binding status of the captcha like this:
```
Checking element binding of _submitForm...
true
Checking element binding of _captchaForm...
true
Checking element binding of _captchaSubmit...
true
```
A validation exception in Laravel will not be recorded in `laravel.log` by default. 

I will suggest you dump your inputs on Laravel to see if you can get `g-recaptcha-response` param. If not, that means your frontend code doesn't work properly. You can also paste your code here for further discussion.

**Comment by david-j-davis** [comment]
Thanks @albertcht. So I still have no idea where to find:
`Checking element binding of _submitForm...
true
Checking element binding of _captchaForm...
true
Checking element binding of _captchaSubmit...
true`

I've dumped out my requests, and my code doesn't make it past the validator. Here's my code, followed by the dump:

` protected function validator(array $data)
    {
        echo dd($data);
        $rules = [
            'name' => 'required|string|max:255|unique:users',
            'email' => 'required|string|email|max:255|unique:users',
            'password' => 'required|string|min:6|confirmed',
        ];

        if (config('auth.captcha.registration')) {
            $rules['g-recaptcha-response'] = 'required|captcha';
        }

        return Validator::make($data, $rules);
    }`

Dump:
`array:7 [▼
  "_token" => "8CbcDV9plLdcZV30VmHvD5bs5kbQqAk4vu0HIk2U"
  "name" => "someguy"
  "email" => "someguy@gmail.com"
  "email_confirmation" => "someguy@gmail.com"
  "password" => ""
  "password_confirmation" => ""
  "g-recaptcha-response" => null
]`

**Comment by albertcht** [comment]
Hi @david-j-davis ,

Please refer to https://developers.google.com/web/tools/chrome-devtools/console/ to open your browser debug console. Go find if there's any error message on your console.

As  what your dump data has shown, `g-recaptcha-response` is null. That's why you can't pass the captcha rule.

Also you can check again if your captcha directive is placed on a correct position. (see: https://github.com/albertcht/invisible-recaptcha#usage)

If the error still goes on, please paste your console error message and your partial code of your blade file as reference, thanks.

**Comment by david-j-davis** [comment]
Hi @albertcht, thanks. I don't have any console error logs. Haven't had any errors posted to the console from the start even though I do have INVISIBLE_RECAPTCHA_DEBUG set to true. When I hit submit, this shows up in the console, and nothing else:
`recaptcha__en.js:84 XHR finished loading: POST "https://www.google.com/recaptcha/api2/reload?k=6LfrujEUAAAAAPC87ZEt-gsms_5ZigErHpju2s2d".`

Here is partial code from my blade file. As a side note I am certain that my site key and secret key are set up correctly in my eng file. Still nothing happens after submit other than the inputs clear and the captcha fails:

` <div class="field">
                <label for="password-confirm" class="label is-medium">Confirm Password</label>

                <div class="control is-expanded has-icons-left">
                    <input id="password-confirm" type="password" class="input is-medium" name="password_confirmation" required>
                    <span class="icon is-small is-left">
                      <i class="fa fa-key"></i>
                    </span>
                </div>
            </div>

            @if(config('auth.captcha.registration'))
                @captcha()
            @endif

            

            <div class="field">
                <p class="control">
                    <button type="submit" class="button is-primary">
                        Register
                    </button>
                </p>
            </div>
        </form>`

**Comment by albertcht** [comment]
Hi @david-j-davis ,

Can you open your browser console and type `_submitForm` and `_captchaForm`? Let's see what it shows.

**Comment by david-j-davis** [comment]
Both are undefined..

![screen shot 2018-03-19 at 6 14 34 pm](https://user-images.githubusercontent.com/3498223/37630402-a20c9a64-2ba1-11e8-87dd-6da8152d2531.png)

**Comment by albertcht** [comment]
Well... it looks like recaptcha's `api.js` loaded successfully, so you can see the badge logo. However, your element bindings are incorrect since you can't get `_captchaForm` and `_submitForm` elements correctly.

You check https://github.com/albertcht/invisible-recaptcha#usage again if these three conditions are satisfied.  And do you mind pasting your source html code(view-source) of that page here?

**Comment by david-j-davis** [comment]
All three conditions seem to be met. Here's my source:

`

  | <!DOCTYPE html>
  | <html lang="en">
  | <head>
  | 
  | <script async src="https://www.googletagmanager.com/gtag/js?id=UA-106769408-1"></script>
  | <script>
  | window.dataLayer = window.dataLayer \|\| [];
  | function gtag(){dataLayer.push(arguments)};
  | gtag('js', new Date());
  | gtag('config', 'UA-106769408-1');
  | </script>
  |  
  | <meta charset="utf-8">
  | <meta http-equiv="X-UA-Compatible" content="IE=edge">
  | <meta name="viewport" content="width=device-width, initial-scale=1">
  |  
  | 
  | <meta name="csrf-token" content="NiLzJLuyDmPkYH7jreGj72ciXlqJLW0PjLf1Kz8N">
  |  
  | <title>MyRecipeZone</title>
  |  
  | 
  | <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico">
  | <link rel="stylesheet" type="text/css" href="https://myrecipe.zone/css/font-awesome.min.css">
  | <link rel="stylesheet" type="text/css" href="/css/app.css">
  |  
  |  
  | </head>
  | <body>
  | <div id="recipe-app">
  | <nav class="navbar">
  | <div class="navbar-brand">
  | <a class="navbar-item" href="/">
  | <img class="logo" src="https://myrecipe.zone/img/logo.svg">
  | </a>
  | <div class="navbar-burger">
  | <span></span>
  | <span></span>
  | <span></span>
  | </div>
  | </div>
  | <div id="navbar" class="navbar-end">
  | <a class="navbar-item" href="https://myrecipe.zone/login">Login</a>
  | <a class="navbar-item" href="https://myrecipe.zone/register">Register</a>
  | </div>
  | </nav>
  | 
  | <section class="section">
  | <div class="container is-fluid register-page">
  | <nav class="panel">
  | <p class="panel-heading">Register</p>
  | <div class="panel-block">
  | <form method="POST" action="https://myrecipe.zone/register">
  | <input type="hidden" name="_token" value="NiLzJLuyDmPkYH7jreGj72ciXlqJLW0PjLf1Kz8N">
  |  
  | <div class="field">
  | <label for="name" class="label is-medium">Username</label>
  | <div class="control is-expanded has-icons-left">
  | <input id="name" type="text" class="input is-medium " name="name" value="" placeholder="Please enter a unique username" required autofocus>
  | <span class="icon is-small is-left">
  | <i class="fa fa-user"></i>
  | </span>
  |  
  | </div>
  | </div>
  |  
  | <div class="field">
  | <label for="email" class="label is-medium">E-Mail Address</label>
  | <div class="control is-expanded has-icons-left">
  | <input id="email" type="email" class="input is-medium " name="email" value="" required>
  | <span class="icon is-small is-left">
  | <i class="fa fa-envelope"></i>
  | </span>
  |  
  | </div>
  | </div>
  |  
  | <div class="field">
  | <label for="email-confirmation" class="label is-medium">Confirm E-Mail</label>
  | <div class="control is-expanded has-icons-left">
  | <input id="email-confirmation" type="email" class="input is-medium" name="email_confirmation" required>
  | <span class="icon is-small is-left">
  | <i class="fa fa-envelope"></i>
  | </span>
  | </div>
  | </div>
  |  
  | <div class="field">
  | <label for="password" class="label is-medium">Password</label>
  | <div class="control is-expanded has-icons-left">
  | <input id="password" type="password" class="input is-medium " name="password" required>
  | <span class="icon is-small is-left">
  | <i class="fa fa-key"></i>
  | </span>
  |  
  | </div>
  | </div>
  |  
  | <div class="field">
  | <label for="password-confirm" class="label is-medium">Confirm Password</label>
  |  
  | <div class="control is-expanded has-icons-left">
  | <input id="password-confirm" type="password" class="input is-medium" name="password_confirmation" required>
  | <span class="icon is-small is-left">
  | <i class="fa fa-key"></i>
  | </span>
  | </div>
  | </div>
  |  
  | <script src="https://cdn.polyfill.io/v2/polyfill.min.js"></script>
  | <div id="_g-recaptcha"></div>
  | <div class="g-recaptcha" data-sitekey="6LfrujEUAAAAAPC87ZEt-gsms_5ZigErHpju2s2d" data-size="invisible" data-callback="_submitForm" data-badge="bottomright"></div><script src="https://www.google.com/recaptcha/api.js" async defer></script>
  | <script>var _submitForm,_captchaForm,_captchaSubmit,_execute=true;</script><script>window.onload=function(){_captchaForm=document.querySelector("#_g-recaptcha").closest("form");_captchaSubmit=_captchaForm.querySelector('[type=submit]');_submitForm=function(){if(typeof _submitEvent==="function"){_submitEvent();grecaptcha.reset();}else{_captchaForm.submit();}};_captchaForm.addEventListener('submit',function(e){e.preventDefault();if(typeof _beforeSubmit==='function'){_execute=_beforeSubmit();}if(_execute){grecaptcha.execute();}});console.log("Checking element binding of _submitForm...");console.log(_submitForm!==undefined);console.log("Checking element binding of _captchaForm...");console.log(_captchaForm!==undefined);console.log("Checking element binding of _captchaSubmit...");console.log(_captchaSubmit!==undefined);}</script>
  |  
  |  
  |  
  | <div class="field">
  | <p class="control">
  | <button type="submit" class="button is-primary">
  | Register
  | </button>
  | </p>
  | </div>
  | </form>
  | </div>
  |  
  | </nav>
  | </div>
  | </section>
  | </div>
  | <footer>
  | <script src="https://myrecipe.zone/js/vendor/ckeditor/ckeditor.js"></script>
  | <script src="/js/app.js"></script>
  | </footer>
  |  
  | </body>
  | </html>
  |  

`

**Comment by david-j-davis** [comment]
Conditions seem to be met. Here's my view source paste: https://paste.ofcode.org/HLcJz8XQMgQA6ubHFhtvic

**Comment by albertcht** [comment]
Hi @david-j-davis ,

I ran your html code on my computer, and this is what I got:

![](https://i.imgur.com/veg4w5I.png)

And I try to dump `g-recaptcha-response` by adding:
```
<script type="text/javascript">
        _submitEvent = function() {
            console.log(document.querySelector("#g-recaptcha-response").value);
        };
</script>
```
And here is the result:

![](https://i.imgur.com/tuo00jC.png)

everything looks good, and I can't find any error here, with the modified html I tested: https://paste.ofcode.org/eDG5sBLsr6Shni9dsuPZkE

**Comment by david-j-davis** [comment]
That's curious. Ok not sure what's going on then on my end. Thanks for
helping me.

**Comment by albertcht** [comment]
No problem.

**Comment by talapai** [comment]
Exactly such a problem. 
I use Cartalyst\Sentinel not official Auth package. (Laravel 5.6. )
"g-recaptcha-response" field is empty. 
_captchaForm and _submitForm are undefined

**Comment by albertcht** [comment]
Hi @talapai ,

I created a demo page here: https://codepen.io/albertcht/pen/yjLoQY

The script inside the form is 100% same with the js this package generated.

If you can re-demo your error on codepen like what I did will help me a lot to figure out what happened. Thank you.

**Comment by albertcht** [comment]
I am currently busy working on another project, but I will check this issue once again. If there's anyone who can reproduce the issue on codepen will help me a lot!

**Comment by johnnymast** [comment]
isnt this this same issue as i issued before? i gave you a 1000 word explaination how to fix it but nothing gets fixed. I think i will just fork the project and release a better version my self.. With your credits ofcource.

**Comment by johnnymast** [comment]
@albertcht im sorry for my tone. I didnt realize it was a bit harsh for that im sorry,

**Comment by ColinTravis** [comment]
Just wanted to add to this thread that the fix in the commit fixed the same issue for me as well. Any way to ensure the fix is in place? When I deploy composer re-downloads my packages.

**Comment by albertcht** [comment]
That commit has been released in `v1.8.3`
